changelog.d/16288.bugfix

```diff
@@ -0,0 +1 @@
+Fix bug when rehydrated devices do not have the device refresh tokens. Contributed by Hanadi.
```

Comment:
Any idea when this bug was introduced?

Reply:
Ok so I did a little digging, seems that dehydrated device where introduced in [v1.22](https://github.com/matrix-org/synapse/releases/tag/v1.22.0) and refresh tokens in [v1.49](https://github.com/matrix-org/synapse/releases/tag/v1.49.0) so I would say this bug would have appeared since v1.49. But only for those unlucky ones who have those two features enabled 😄 
Would you have a suggestion on what I could add to the change log message regarding that?

Reply:
```suggestion
Fix bug introduced in Synapse 1.49.0 when using dehydrated devices ([MSC2697](https://github.com/matrix-org/matrix-spec-proposals/pull/2697)) and refresh tokens. Contributed by Hanadi.
```